src/vibrator-sysfs.cpp

```diff
@@ -34,7 +34,12 @@ VibratorSysfs::VibratorSysfs(): Vibrator() {
 
     m_device = udev.device_from_syspath("/sys/class/leds/vibrator");
 
-    m_device.set_sysattr("trigger", "transient");
+    try {
+        m_device.set_sysattr("trigger", "transient");
+    } catch (const std::runtime_error& e) {
+        (void)e;
+        std::cerr << "Failed to add transient trigger, usually non-fatal" << std::endl;
```

Comment:
Is the wording correct here? I tried to understand what it does, and found https://01.org/linuxgraphics/gfx-docs/drm/leds/ledtrig-transient.html. Maybe it should be "failed to enable transient trigger"?

Reply:
I think its ok. Not many ppl will have this error hopefully, and if yes they need to look in the code anyways, from the error msg alone you would not know where this is coming from.

Reply:
Problem might occur later when we also try to set `duration` since thats a property of transient, problem nr 2 is that some android kernels is not standard and implement this regardless of transient is set or not. So i would like to have a "try" on the set_sysattr bellow too. Or maybe we check if the sysattr exists before continuing using has_sysattr(). Since it might also abort if state, duration or activate does not exist also. See configure() bellow.

We should do these checks in usable() so we use dummyVibrator if something here goes wrong, we should never abort or crash. I might also rework the usable() part to use throw and catch to check if a device is "usable"

Reply:
I've reworked the detection mechanism.